File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/model/response/UserResponse.java
package com.aitor.api_tfg.model.response;

import com.aitor.api_tfg.model.db.Gender;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserResponse {
    private Integer id;
    private String image;
    private String Username;
    private String nombre;
    private String apellidos;
    private String email;
    private Double peso;
    private Integer altura;
    private Gender genero;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/perfil/data/response/UserResponse.kt
package com.aitor.trackactividades.perfil.data.response

import com.aitor.trackactividades.core.model.Gender
import com.aitor.trackactividades.perfil.presentation.model.UserModel
import kotlinx.serialization.Serializable

@Serializable
data class UserResponse(
    val id: Int,
    val image: String?,
    val username: String,
    val nombre: String,
    val apellidos: String,
    val email: String,
    val peso: Double,
    val altura: Int,
    val genero: Gender
) {
    override fun toString(): String {
        return """
            UserResponse(
                id=$id,
                image=$image,
                username='$username',
                nombre='$nombre',
                apellidos='$apellidos',
                email='$email',
                peso=$peso,
                altura=$altura,
                genero='$genero'
            )
        """.trimIndent()
    }

    fun toPresentation(): UserModel {
        return UserModel(
            id = id,
            image = image,
            username = username,
            nombre = nombre,
            apellidos = apellidos,
            email = email,
            peso = peso,
            altura = altura,
            genero = genero
        )
    }
}

